pkg/plugin/run.go

```diff
@@ -109,7 +107,17 @@ func (e *AutoscaleEnforcer) startPermitHandler(ctx context.Context) error {
 }
 
 // Returns body (if successful), status code, error (if unsuccessful)
-func (e *AutoscaleEnforcer) handleAgentRequest(req api.AgentRequest, nodeName *string) (*api.PluginResponse, int, error) {
+func (e *AutoscaleEnforcer) handleAgentRequest(
+	req api.AgentRequest,
+) (_ *api.PluginResponse, status int, _ error) {
+	nodeName := "<none>" // override this later if we have a node name
+	defer func() {
+		hasMetrics := req.Metrics != nil
+		e.metrics.validResourceRequests.
```

Comment:
shouldn't we not log if we don't have metrics? 

What, operationally, do we learn from being able to distinguish metrics-ey requests from not-metricsy requests?

(I believe that the size of the metrics payload is (roughly) constant with regards to the number of timeseries (metrics*labels) even if storage is delta and zero run length encoded.)